### PR TITLE
Various migration features and fixes

### PR DESCRIPTION
**Pull request by alexandraorth**
+ Blocking loader fix
+ New CSV parser
+ Split macro
+ Migration prints number of jobs completed

### CODE REVIEW

File: grakn-engine/src/main/java/ai/grakn/engine/backgroundtasks/InMemoryTaskManager.java [hunk 1]
```diff
@@ -122,8 +122,14 @@ public CompletableFuture completableFuture(String taskId) {
...
             return CompletableFuture.runAsync(() -> {
                 try {
-                    instantiatedTasks.get(taskId).getKey().get();
-                } catch (InterruptedException | ExecutionException e) {
+                    while(true){
+                        TaskState state = storage().getState(taskId);
+                        if(state.status().equals(COMPLETED) || state.status().equals(FAILED)){
+                            break;
+                        }
+                        Thread.sleep(1000);
```
Comment (fppt) [inline grakn-engine/src/main/java/ai/grakn/engine/backgroundtasks/InMemoryTaskManager.java]: Why is the random sleep needed now ?

File: grakn-graql/src/main/java/ai/grakn/graql/QueryBuilderImpl.java [hunk 2]
```diff
@@ -189,7 +189,8 @@ public Pattern parsePattern(String patternString) {
      */
     @Override
     public <T extends Query<?>> T parseTemplate(String template, Map<String, Object> data){
-        return parse(templateParser.parseTemplate(template, data));
+        String templated = templateParser.parseTemplate(template, data);
```
Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/QueryBuilderImpl.java]: Inline `templated` no need for a variable here.

File: grakn-test/src/test/java/ai/grakn/test/graql/template/TemplateParserTest.java [hunk 3]
```diff
@@ -58,7 +71,7 @@ public void multiValueOneLineTest(){
         assertParseEquals(template, data, expected);
     }
 
-    @Test(expected = RuntimeException.class)
+    @Test(expected = GraqlTemplateParsingException.class)
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/template/TemplateParserTest.java]: Looking into using `ExpectedException` instead. It lets you check the contents of the error message as well. It's a more accurate check.

Comment (alexandraorth) [inline grakn-test/src/test/java/ai/grakn/test/graql/template/TemplateParserTest.java]: Yep, I do that in my migration tests and should here as well.

File: grakn-graql/src/main/java/ai/grakn/graql/QueryBuilderImpl.java (hunk 2 above)
Comment (alexandraorth) [inline grakn-graql/src/main/java/ai/grakn/graql/QueryBuilderImpl.java]: Will change.

**Pull request merged**